pkgs/by-name/s7/s7/package.nix

```diff
@@ -0,0 +1,159 @@
+{
+  lib,
+  fetchFromGitLab,
+  stdenv,
+  notcurses,
+  gmp,
+  mpfr,
+  libmpc,
+  flint3,
+  unstableGitUpdater,
+  writeScript,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "s7";
+  version = "11.2-unstable-2024-11-02";
+
+  src = fetchFromGitLab {
+    domain = "cm-gitlab.stanford.edu";
+    owner = "bil";
+    repo = "s7";
+    rev = "6b6651de318c3360aa2711a445659f372ffe894e";
+    hash = "sha256-9x1u60jflESAHgP0gwIDVFwhI5XU5gsAiXaic8zdW2U=";
+  };
+
+  buildInputs = [
+    notcurses
+    gmp
+    mpfr
+    libmpc
+    flint3
+  ];
+
+  # The following scripts are modified from [Guix's](https://packages.guix.gnu.org/packages/s7/).
+
+  patchPhase = ''
+    runHook prePatch
+
+    substituteInPlace s7.c \
+        --replace-fail libc_s7.so $out/lib/libc_s7.so
+
+    runHook postPatch
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    # The older REPL
+    cc s7.c -o s7-repl \
+        -O2 -I. \
+        -Wl,-export-dynamic \
+        -lm -ldl \
+        -DWITH_MAIN \
+        -DS7_LOAD_PATH=\"$out/share/s7/scm\"
+
+    # The newer REPL powered by Notcurses
+    cc s7.c -o s7-nrepl \
+        -O2 -I. \
+        -Wl,-export-dynamic \
+        -lm -ldl \
+        -DWITH_MAIN \
+        -DWITH_NOTCURSES -lnotcurses-core \
+        -DS7_LOAD_PATH=\"$out/share/s7/scm\"
+
+    cc libarb_s7.c -o libarb_s7.so \
+        -O2 -I. \
+        -shared \
+        -lflint -lmpc \
+        -fPIC
+
+    cc notcurses_s7.c -o libnotcurses_s7.so \
+        -O2 -I. \
+        -shared \
+        -lnotcurses-core \
+        -fPIC
+
+    cc s7.c -c -o s7.o \
+        -O2 -I. \
+        -ldl -lm
+
+    cc ffitest.c s7.o -o ffitest \
+        -I. \
+        -Wl,-export-dynamic \
+        -ldl -lm
+
+    ./s7-repl libc.scm
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    dst_bin=$out/bin
+    dst_lib=$out/lib
+    dst_inc=$out/include
+    dst_share=$out/share/s7
+    dst_scm=$out/share/s7/scm
+    dst_doc=$out/share/doc/s7
+    mkdir -p $dst_bin $dst_lib $dst_inc $dst_share $dst_scm $dst_doc
+
+    cp -pr -t $dst_bin s7-repl s7-nrepl
+    ln -s s7-nrepl $dst_bin/s7
+    cp -pr -t $dst_lib libarb_s7.so libnotcurses_s7.so libc_s7.so
+    cp -pr -t $dst_share s7.c
+    cp -pr -t $dst_inc s7.h
+    cp -pr -t $dst_scm *.scm
+    cp -pr -t $dst_doc s7.html
+
+    runHook postInstall
+  '';
+
+  doInstallCheck = true;
+
+  installCheckPhase = ''
+    runHook preInstallCheck
+
+    $dst_bin/s7-repl s7test.scm
```

Comment:
I am a bit late to the party, but I find that even though there are errors the test still succeed.

```console
Running phase: installCheckPhase
@nix { "action": "setPhase", "phase": "installCheckPhase" }
s7: 4-Nov-2024
load s7test.scm
ffitest sh: ffitest: not found



;append second argument, #f, is boolean but should be a string
;    (append "/home/" username "/cl/")
;    s7test.scm, line 20439, position: 850325
; (append "/home/" username "/cl/")
; ((test *cload-directory* (append "/home/"... ; old-dir: ""
```

Reply:
I think it is because the `ffitest` was not moved to `$out/bin` when I refactored this from `checkPhase` to `installCheckPhase`. 😕